node/malus/src/variants/common.rs

```diff
@@ -0,0 +1,347 @@
+		persisted_validation_data,
+	));
+
+	gum::debug!(
+		target: MALUS,
+		para_id = ?candidate_descriptor.para_id,
+		"ValidationResult: {:?}",
+		&result
+	);
+
+	response_sender.send(result).unwrap();
+}
+
+// TODO: Only fake validation for `MALICIOUS_POV`, otherwise behave normally.
+impl<Sender, Spawner> MessageInterceptor<Sender> for ReplaceValidationResult<Spawner>
+where
+	Sender: overseer::SubsystemSender<CandidateValidationMessage>
+		+ overseer::SubsystemSender<AllMessages>
+		+ Clone
+		+ Send
+		+ 'static,
+	Spawner: SpawnNamed + Clone + 'static,
+{
+	type Message = CandidateValidationMessage;
+
+	// Capture all candidate validation requests and depending on configuration fail them.
+	fn intercept_incoming(
+		&self,
+		subsystem_sender: &mut Sender,
+		msg: FromOverseer<Self::Message>,
+	) -> Option<FromOverseer<Self::Message>> {
+		match msg {
+			FromOverseer::Communication {
+				msg:
+					CandidateValidationMessage::ValidateFromExhaustive(
+						validation_data,
+						validation_code,
+						descriptor,
+						pov,
+						timeout,
+						sender,
+						commitments_hash,
+					),
+			} => {
+				match self.fake_validation {
+					FakeCandidateValidation::ApprovalValid |
+					FakeCandidateValidation::BackingAndApprovalValid => {
+						// Behave normally if the `PoV` is not known to be malicious.
+						if pov.block_data.0.as_slice() != MALICIOUS_POV {
+							return Some(FromOverseer::Communication {
+								msg: CandidateValidationMessage::ValidateFromExhaustive(
+									validation_data,
+									validation_code,
+									descriptor,
+									pov,
+									timeout,
+									sender,
+									commitments_hash,
+								),
+							})
+						}
+						create_validation_response(validation_data, descriptor, pov, sender);
+						None
+					},
+					FakeCandidateValidation::ApprovalInvalid |
+					FakeCandidateValidation::BackingAndApprovalInvalid => {
+						let validation_result =
+							ValidationResult::Invalid(InvalidCandidate::InvalidOutputs);
+
+						gum::info!(
+							target: MALUS,
+							para_id = ?descriptor.para_id,
+							"ValidateFromExhaustive result: {:?}",
+							&validation_result
+						);
+						// We're not even checking the candidate, this makes us appear faster than honest validators.
+						sender.send(Ok(validation_result)).unwrap();
+						None
+					},
+					_ => Some(FromOverseer::Communication {
+						msg: CandidateValidationMessage::ValidateFromExhaustive(
+							validation_data,
+							validation_code,
+							descriptor,
+							pov,
+							timeout,
+							sender,
+							commitments_hash,
+						),
+					}),
+				}
+			},
+			FromOverseer::Communication {
+				msg:
+					CandidateValidationMessage::ValidateFromChainState(
+						descriptor,
+						pov,
+						timeout,
+						response_sender,
+						commitments_hash,
+					),
+			} => {
+				match self.fake_validation {
+					FakeCandidateValidation::BackingValid |
+					FakeCandidateValidation::BackingAndApprovalValid => {
```

Comment:
This is because this message is sent by the backing subsystem only. There is an identical check for the other susbsytem message sent by approval voting where check ApprovalValid | BackingAndApprovalValid for the same reason.